flare/gp_algebra.py

```diff
@@ -1719,6 +1819,78 @@ def get_like_from_mats(ky_mat, l_mat, alpha, name):
     return like
 
 
+def get_neg_like(
+    hyps: np.ndarray,
+    name: str,
+    force_kernel,
+    logger_name=None,
+    cutoffs=None,
+    hyps_mask=None,
+    n_cpus=1,
+    n_sample=100,
+):
+    """compute the log likelihood and its gradients
+    :param hyps: list of hyper-parameters
+    :type hyps: np.ndarray
+    :param name: name of the gp instance.
+    :param kernel_grad: function object of the kernel gradient
+    :param output: Output object for dumping every hyper-parameter
+                   sets computed
+    :type output: logger_name
+    :param cutoffs: The cutoff values used for the atomic environments
+    :type cutoffs: list of 2 float numbers
+    :param hyps_mask: dictionary used for multi-group hyperparmeters
+    :param n_cpus: number of cpus to use.
+    :param n_sample: the size of block for matrix to compute
+    :return: float, np.array
+    """
+
+    time0 = time.time()
+
+    ky_mat = get_force_block(
+        hyps,
+        name,
+        force_kernel,
+        cutoffs=cutoffs,
+        hyps_mask=hyps_mask,
+        n_cpus=n_cpus,
+        n_sample=n_sample,
+    )
+
+    logger = logging.getLogger(logger_name)
+    logger.debug(f"{name} get_ky_and_hyp {time.time()-time0}")
+
+    time0 = time.time()
+
+    # catch linear algebra errors
+    try:
+        ky_mat_inv = np.linalg.inv(ky_mat)
+        l_mat = np.linalg.cholesky(ky_mat)
+    except np.linalg.LinAlgError:
+        number_of_hyps = len(hyps)
+        return -1e8, np.zeros(number_of_hyps)
+
+    labels = _global_training_labels[name]
+
+    alpha = np.matmul(ky_mat_inv, labels)
+    alpha_mat = np.matmul(alpha.reshape(-1, 1), alpha.reshape(1, -1))
+    like_mat = alpha_mat - ky_mat_inv
```

Comment:
I remember writing this function, but don't remember why I didn't use get_like_from_mat for the following lines... maybe double check whether the codes are repeated from line 1878 to the end of the function

Reply:
Good catch, and I fixed the repeated code. Looks like alpha mat and like mat aren't used.